### ISSUE TITLE
SearchRequestBuilder.setHighlighterBoundaryChars doesn't work as expected

### ISSUE THREAD

**Issue by julamb** [elastic/elasticsearch#15792]
Using the java API, I'm trying to use setHighlighterBoundaryChars to change the boundary chars of the fast vector highlighter, but it seems that it is broken.

```
...
char[] chars = new char[]{'\n', '.'};
request.setHighlighterBoundaryChars(chars);
System.out.println(request.toString());
...
```

prints 

````
...
  "highlight" : {
    "boundary_max_scan" : 200,
    "boundary_chars" : "[C@550ec6a1",
...
```

According to [the doc](https://www.elastic.co/guide/en/elasticsearch/reference/2.0/search-request-highlighting.html#boundary-characters), the "boundary_chars" field should be a string containing all the boundary chars.


###### Possible fix
Looking at the code of HighlightBuilder.java:368 in 2.2: 
```
// Current
builder.field("boundary_chars", boundaryChars); // boundaryChars is a char array

// Fixed
builder.field("boundary_chars", new String(boundaryChars)); // might work better
```

**Comment by nik9000** [comment]
Confirmed the bug. It looks like your fix will work too. This is not a problem in master though. I'll open up a PR to fix it once the test suite finishes running.

**Comment by julamb** [comment]
Cool! Actually I just noticed that there's another use of the boundaryChars char array at HighlightBuilder.java:430.

**Comment by nik9000** [comment]
> Cool! Actually I just noticed that there's another use of the boundaryChars char array at HighlightBuilder.java:430.

Good catch. I'll fix that one too.

**Comment by nik9000** [comment]
> Good catch. I'll fix that one too.

Done and waiting in the PR.

**Comment by clintongormley** [comment]
Fixed by #15795
